include/sentry.h

```diff
@@ -1362,6 +1362,14 @@ SENTRY_API sentry_user_consent_t sentry_user_consent_get(void);
  */
 SENTRY_API sentry_uuid_t sentry_capture_event(sentry_value_t event);
 
+struct sentry_path_s;
+typedef struct sentry_path_s sentry_path_t;
+
+/**
+ *  This function allows for independently created minidumps to be captured.
+ */
+SENTRY_API void sentry_capture_minidump(sentry_path_t *dump_path, sentry_value_t event, const sentry_options_t *options);
```

Comment:
The usage of `sentry_options_t` in the public interface after the backend's initialization seems unusual. For instance, the SDK for Unreal Engine does not maintain an instance of its own. This appears to align with `sentry_init()`, which takes ownership of the object.

Reply:
Changed all that on my next commit. Essentially im trying to mirror what is in the breakpad backend without platform specific API stuff. Mainly relating to ```on_crash_func``` where in the breakpad backend it uses ```EXCEPTION_POINTERS *exinfo``` but only on windows. I am hoping to have the breakpad backend call into the new function to minimize duplicate code.

Reply:
The critical part is that options are sunk into the SDK when `sentry_init()` is called. Access to it is unsafe after `sentry_init()`, and all functions requiring access to the global state should do so internally via the `SENTRY_WITH_OPTIONS` scope macro, which handles synchronized access to the ref-count.

Reply:
> Essentially im trying to mirror what is in the breakpad backend without platform specific API stuff. Mainly relating to on_crash_func where in the breakpad backend it uses EXCEPTION_POINTERS *exinfo but only on windows.

These shouldn't be necessary to touch for the function at all. Only the stuff inside the `should_dump` conditional (and the `SENTRY_WITH_OPTIONS` access to options) is essential.

Reply:
> I am hoping to have the breakpad backend call into the new function to minimize duplicate code.

Let's leave deduplication open until the PR is done. There is a good chance that these implementations will diverge down the road, so there is no immediate need to keep one bound to the other.